contents/javascript/Javascript-framework-React.md

```diff
@@ -0,0 +1,140 @@
+
+Multiple components can be used to compose complex user interfaces. For example, if you have a todo list application, it could have a `Form` component to handle adding new items, and a `List` component to display the items.
+Furthermore, the `List` component could be composed of multiple `ListItem` components, one for each item in the todo list. 
+
+The sandbox below has a simple example to show you how components work:
+
+<box type="info">
+    Notice the HTML-like syntax within the JavaScript method <code>render()</code>. This is <a href="https://reactjs.org/docs/introducing-jsx.html">JSX</a>, and it is a JavaScript syntax extension developed by the React team.
+</box>
+
+<iframe src="https://codesandbox.io/embed/884m88wr90?fontsize=12" style="width:100%; height:400px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+<br/>
+
+With components written in JavaScript, you can easily pass data through your application and keep state out of the DOM. You can read more about components and props in the React [documentation](https://reactjs.org/docs/components-and-props.html). 
+
+#### React is Fast
+
+Web applications can have a lot of user interaction and data updates. Extensive DOM manipulation can be a performance bottleneck in these applications as the DOM is tree-structured - simple changes at the top can trigger the entire application to get updated. 
+
+React solves this problem by using a virtual DOM. The virtual DOM is a JavaScript object that is kept in the memory of your application. In other words, it is a virtual representation of the actual DOM.
+
+Any updates to your application will first be made to the virtual DOM. Then, React will compare the virtual DOM with the actual DOM using a [diffing algorithm](https://reactjs.org/docs/reconciliation.html#the-diffing-algorithm) to find out what changes have to be made to the actual DOM. 
+Finally, React figures out how to update the actual DOM in the most effective way possible to guarantee minimum update time. This results in better performance and a cleaner user experience, even with data-intensive applications.
+
+#### Other Advantages of React
+
+Besides the three main benefits explained above, React has the following advantages:
+ - React only allows data to flow downwards (one-way data binding), which makes your application easier to debug.
+ - As it is an open-source library backed by Facebook, it is constantly being updated with new features and performance optimizations.
+ - If you want to manage the state of your application using [Redux](https://redux.js.org/introduction/getting-started), React might be a better choice than other libraries / frameworks as there are more resources for learning React + Redux.
+
+### When to not use React
```

Comment:
The points below do not match this title. You'll need to either change this title or restructure the way the points are presented